### DISCUSSION TITLE
Comunication between Host and Docker Containner using FIFO pipes by bind mount (Linux)

### DISCUSSION THREAD

**Discussion by shamirtahir50** [moby/moby#45758] [General]
My container creates fifo pipe in the bind mounted dir but container when reads or writes to this pipe, the host cant recieve it or vice versa. with similar permissions to create or open pipe works for comunication within host and within containner. But it does not work in between host and containner.

The pipe is created and reads/writes in cpp program within containeer. Pipe created with both 666 and 777 permisssions and opened with O_RDWR. Simmilar cpp programs reads/writes in the host. Again these program works fine within containner and within host the problem happens only in host communication.

docker run -it --name broken-container -v /tmp:/app2 cppdocker
send.cpp Dockerfile just compiles it.
#include <iostream> #include <cstring> #include <fcntl.h> #include <sys/stat.h> #include <sys/types.h> #include <unistd.h> int main() { int fd; const char* pipeName = "/app2/my_pipe"; // Pipe file path fd = open(pipeName, O_RDWR); char * message = "hello"; int check = write(fd, message, strlen(message) + 1); close(fd); return 0; }

recieve.cpp runs on host 
#include <iostream> #include <cstring> #include <fcntl.h> #include <sys/stat.h> #include <sys/types.h> #include <unistd.h> int main() { const char* pipeName = "/tmp/my_pipe"; int val = mkfifo(pipeName, 0777); char buffer[80]; int fd = open(pipeName, O_RDWR); ssize_t bytesRead = read(fd, buffer, sizeof(buffer) - 1); buffer[bytesRead] = '\0'; close(fd); std::cout << "Received message: " << buffer << std::endl; return 0; }

**Comment by thaJeztah** [comment]
What platform are you running on? Is this on Linux, are you running Docker Desktop?

**Reply by shamirtahir50** [reply]
@thaJeztah I am running it on Debian using Docker Engine I have uninstalled the Docker Desktop Now running Container in Docker Engine but still no communication is being done through fifo pipe

**Reply by arothmel** [reply]
Is it a SELinux issue? 

docker run -it --name broken-container -v /tmp:/app2:Z cppdocker

[Modify volume mounts SELinux labels on the fly based on :Z or :z](https://github.com/rhatdan/moby1/commit/e6473011583967df4aa5a62f173fb421cae2bb1e)

**Reply by shamirtahir50** [reply]
In my case the issue was with pipe permissions. I could be resolved by using chmod while creating pipe or assigining the user and group of host to containeer so that the containner have permession to write or read the pipe,

docker run -it --name broken-container -v /tmp:/app2 -u user:group cppdocker
